### PR TITLE
MRI VM Stats

### PR DESCRIPTION
**Pull request by jeffkreeftmeijer**
This patch adds Ruby VM stats to be used with a Magic Dashboard. It reports allocated objects, garbage collection run stats, heap slots, stats from RubyVM.stats, thread counts, and total GC time, because those seem to be the most reported. Let me know if there’s something you’re missing for these!

Please help me test this by running a Ruby app (any one from the test setups will do) locally on this version of the Ruby integration on as many different Ruby versions you can. Also, check out the [dashboard](https://github.com/appsignal/public_config/pull/37) to see if everything there makes sense and properly works. It hasn’t been merged yet, but you can import the serialized version in that pull request.

Feel free to reach out if you need help! :)

### CODE REVIEW

**Review by thijsc — APPROVED** [review]
Really cool! Guess it needs a rebase?

**Comment by jeffkreeftmeijer** [comment]
@thijsc this pull request is up to date. The build is failing because this pull request includes support for Ruby 3.2, and thus that’s added to the build matrix. The grape test is part of #840, which we’re working on. Cherry-picking https://github.com/appsignal/appsignal-ruby/commit/250b36b8c06b9e2d6308b12bd24a1f2a9d477406 in would resolve this, but I’d rather not merge this with the changes we’re making for 3.2. 

We can test this pull request now, while we’re still waiting for the pull request for #840 to be completed. I’ll make sure the build is green before merging.

**Review by tombruijn — APPROVED** [review]

File: lib/appsignal/probes/mri.rb
Comment (unflxw) [inline lib/appsignal/probes/mri.rb]: This is just about naming, so feel free to ignore. I'm also not sure if there's a convention that we usually follow for the metric names.

It seems to me that, for the `heap_slots` distribution value, the metric could just be called `live` or `free`, without the `heap_` prefix. Similarly for the `_gc_count` suffix for the metrics for the `gc_count` distribution value.

File: .changesets/add-a-mri-vm-probe-to-add-a-magic-dashboard.md
Comment (unflxw) [inline .changesets/add-a-mri-vm-probe-to-add-a-magic-dashboard.md:6]: Words!

```suggestion
Add tracking of thread counts, garbage collection runs, heap slots and other garbage collection stats to the default MRI probe. These metrics will be shown in AppSignal.com in a new Ruby VM Magic Dashboard.
```

**Review by unflxw — APPROVED** [review]

**Comment by jeffkreeftmeijer** [comment]
Pushed to `vm-stats` to remove the Ruby 3.2 check I added, as that’s also done in #840. That should get the build green, so we can merge this and #840 after.

File: lib/appsignal/probes/mri.rb
Comment (unflxw) [reply lib/appsignal/probes/mri.rb]: 🤷 Up to you! What you said does seem to me like a good reason to keep them this way, so that they're familiar to users who know Ruby GC stuff and what the values for these keys mean.

**Comment by unflxw** [comment]
Is this good to merge @jeffkreeftmeijer? There's no hurry, but ping me when it's merged, so I can rebase #840 on top and merge that as well. 🙏

**Comment by jeffkreeftmeijer** [comment]
Yes, merging this now!

**Pull request merged**
